docs/user/guides/setup/monorepo.rst

```diff
@@ -0,0 +1,92 @@
+
+.. seealso::
+
+   `sphinx-multiproject <https://sphinx-multiproject.readthedocs.io/en/latest/>`__
+       If you are only using Sphinx projects and want to share the same build configuration,
+       you can also use the ``sphinx-multiproject`` extension.
+
+   :doc:`/guides/environment-variables`
+       You might also be able to reuse the same configuration file across multiple projects,
+       using only environment variables.
+       This is possible if the configuration pattern is very similar and the documentation tool is the same.
+
+Implementation considerations
+-------------------------------
+
+This feature is currently *project-wide*.
+A custom build configuration file path is applied to all versions of your documentation.
+
+.. warning::
+
+   Changing the configuration path will apply to all versions.
+   Different versions of the project may not be able to build again if this path is changed.
+
+Adding an additional project from the same repository
+-----------------------------------------------------
+
+Once you have added the first project from the :ref:`Import Wizard <intro/import-guide:Automatically import your docs>`,
+it will show as if it has already been imported and cannot be imported again.
+In order to add another project with the same repository,
+you will need to use the :ref:`Manual Import <intro/import-guide:Manually import your docs>`.
+
+Setting the custom build configuration file
+-------------------------------------------
+
+Once you have added a Git repository to a project that needs a custom configuration file path,
+navigate to :menuselection:`Admin --> Advanced Settings` and add the path to the :guilabel:`Build configuration file` field.
+
+.. image:: /img/screenshot-howto-build-configuration-file.png
+   :alt: Screenshot of where to find the :guilabel:`Build configuration file` setting.
+
+After pressing :guilabel:`Save`,
+you need to ensure that relevant versions of your documentation are built again.
+
+.. tip::
+
+   Having multiple different build configuration files can be complex.
+   We recommend setting up 1-2 projects in your Monorepo and getting them to build and publish successfully before adding additional projects to the equation.
+
+Next steps
+----------
+
+Once you have your monorepo pattern implemented and tested and it's ready to roll out to all your projects,
+you should also consider the Read the Docs project setup for these individual projects.
+
+Having individual projects gives you the full flexibility of the Read the Docs platform to make individual setups for each project.
+
+For each project, it's now possible to configure:
+
+* Sets of maintainers (or :doc:`organizations </commercial/organizations>` on |com_brand|)
+* :doc:`Custom redirect rules </guides/custom-domains>`
+* :doc:`Custom domains </guides/custom-domains>`
+* :doc:`Automation rules </automation-rules>`
+* :doc:`Traffic and search analytics </reference/analytics>`
```

Comment:
It would be good to mention that different doctools can be used here as well. This is gonna be a key feature for this, I guess.

Reply:
I mentioned it now :+1: 

It would be a lot easier to mention support for generic doctools (static site generators) quickly if we had a good article about specifically "How to use any documentation tool" or something like that.

I referenced our build customization, but I think we could really benefit from having something much more generic?

The how-to could contain example code or references to existing example projects... but that could be developed over time.